Blog post for sprint 190 (#501)
openSUSE/obs-landing: _posts/development/2025-06-02-request-redesign.md

---
layout: post
title:  Improvements To RPM Lint Results and Reviewing Submit Requests
category: development
---

In this update, we’re introducing two improvements that make RPM lint results and reviewing Submit Requests easier: a dedicated view for RPM lint results, and a clearer, more helpful display of build statuses in Submit Requests.

{% include partials/_series-of-posts-about-request-redesign.md %}

The request redesign is part of [the beta program](/2018/10/04/the-beta-program/).

## A New Place For RPM Lints

Previously we were showing the RPM lint results in a small box next to the build results, which had the unfortunate outcome of cramming dense logs into a very small area, and as a tab in the Submit Request, with no connection to a specific build result.

> We now show the RPM lint results in its own page under a new tab called RPM Lint in the package show page. This has the benefit of having a whole page to render the RPM lint. Then on a Submit Request, we link to this tab from each specific build result, sending you directly to the relevant RPM Lint. No need to find your way through the repository and architecture dropdown selectors.

Reviewer: ```suggestion
We now display RPM lint results on a dedicated page under a new 'RPM Lint' tab within the package show page. This has the benefit of having a whole page to render the RPM lint. Then, on a Submit Request, we link to this tab from each specific build result, sending you directly to the relevant RPM Lint. No need to find your way through the repository and architecture dropdown selectors.
```

Author: Why `display` and not `show`? :thinking:

Reviewer: I usually opt for "display" because it's more impersonal and "show" is more an action done by someone.
However, in this case, the sentence starts with "We now show..." so is someone involved anyway... 

So there is no need to change it. Please forget about the show/display stuff, but you can take the other corrections in this paragraph into account as IMO it's easier to understand.

Author: > I usually opt for "display" because it's more impersonal and "show" is more an action done by someone. However, in this case, the sentence starts with "We now show..." so is someone involved anyway...

I like to write things from our team's point of view, so other people see there are human beings after OBS. I'm not a very big fan of writing thing impersonal...

Reviewer: great! I'm fine with "show" in this case.

<figure>
  <img src="/images/posts/sprint-190/2025-05-28-a-new-place-for-rpm-lint.png" alt="A new place for RPM lint">
  <figcaption>A new place for RPM lint</figcaption>
</figure>

## Find Faster If A Submit Request Builds Or Not

> We've changed how we render the build results when a Submit Request is open. The first build results that you'll find, are the ones from the repositories of the target package, because if that's not building it's very likely the Submit Request will not be accepted, right?

Reviewer: ```suggestion
To find out faster if a Submit Request builds or not we've changed how we render the build results when a Submit Request is open. The first build results that you'll find, are the ones from the repositories of the target package, because if that's not building it's very likely the Submit Request will not be accepted, right?
```
